bgpd/bgp_keepalives.c

```diff
@@ -311,8 +311,11 @@ int bgp_keepalives_stop(struct frr_pthread *fpt, void **result)
 {
 	assert(fpt->running);
 
-	atomic_store_explicit(&fpt->running, false, memory_order_relaxed);
-	bgp_keepalives_wake();
```

Comment:
We might drop `bgp_keepalives_wake()` at all. Just changing `bgp_keepalives_on()` to use the same mutex once, instead of twice:

```c
	frr_with_mutex (peerhash_mtx) {
		holder.peer = peer;
		if (!hash_lookup(peerhash, &holder)) {
			struct pkat *pkat = pkat_new(peer);
			(void)hash_get(peerhash, pkat, hash_alloc_intern);
			peer_lock(peer);
		}
		SET_FLAG(peer->thread_flags, PEER_THREAD_KEEPALIVES_ON);
	}
	bgp_keepalives_wake();
```
to:
```c
	frr_with_mutex (peerhash_mtx) {
		holder.peer = peer;
		if (!hash_lookup(peerhash, &holder)) {
			struct pkat *pkat = pkat_new(peer);
			(void)hash_get(peerhash, pkat, hash_alloc_intern);
			peer_lock(peer);
		}
		SET_FLAG(peer->thread_flags, PEER_THREAD_KEEPALIVES_ON);
                pthread_cond_signal(peerhash_cond);
	}
```